providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java

```diff
@@ -129,52 +183,100 @@ public List<Hook> getProviderHooks() {
     ) {
-        return resolveEvaluationGoFeatureFlagProxy(key, defaultValue, evaluationContext, String.class);
+        return getEvaluation(key, defaultValue, evaluationContext, String.class);
     }
 
     @Override
     public ProviderEvaluation<Integer> getIntegerEvaluation(
             String key, Integer defaultValue, EvaluationContext evaluationContext
     ) {
-        return resolveEvaluationGoFeatureFlagProxy(key, defaultValue, evaluationContext, Integer.class);
+        return getEvaluation(key, defaultValue, evaluationContext, Integer.class);
     }
 
     @Override
     public ProviderEvaluation<Double> getDoubleEvaluation(
             String key, Double defaultValue, EvaluationContext evaluationContext
     ) {
-        return resolveEvaluationGoFeatureFlagProxy(key, defaultValue, evaluationContext, Double.class);
+        return getEvaluation(key, defaultValue, evaluationContext, Double.class);
     }
 
     @Override
     public ProviderEvaluation<Value> getObjectEvaluation(
             String key, Value defaultValue, EvaluationContext evaluationContext
     ) {
-        return resolveEvaluationGoFeatureFlagProxy(key, defaultValue, evaluationContext, Value.class);
+        return getEvaluation(key, defaultValue, evaluationContext, Value.class);
+    }
+
+    private String buildCacheKey(String key, String userKey) {
+        return key + "," + userKey;
+    }
+
+    @Override
+    public void initialize(EvaluationContext evaluationContext) throws Exception {
+        FeatureProvider.super.initialize(evaluationContext);
+    }
+
+    private <T> ProviderEvaluation<T> getEvaluation(
+            String key, T defaultValue, EvaluationContext evaluationContext, Class<?> expectedType) {
+        ProviderEvaluation res = null;
+        GoFeatureFlagUser user = GoFeatureFlagUser.fromEvaluationContext(evaluationContext);
+        if (cache == null) {
+            res = resolveEvaluationGoFeatureFlagProxy(key, defaultValue, user, expectedType);
+        } else {
+            res = getProviderEvaluationWithCheckCache(key, defaultValue, expectedType, user);
+        }
+        eventsPublisher.add(Event.builder()
+            .key(key)
+            .defaultValue(defaultValue)
+            .variation(res.getVariant())
+            .value(res.getValue())
+            .userKey(user.getKey())
+            .creationDate(System.currentTimeMillis())
+            .build()
+        );
+        return res;
+    }
+
+    private <T> ProviderEvaluation getProviderEvaluationWithCheckCache(
+            String key, T defaultValue, Class<?> expectedType, GoFeatureFlagUser user) {
+        ProviderEvaluation res = null;
+        try {
+            String cacheKey = buildCacheKey(key, BeanUtils.buildKey(user));
+            res = cache.getIfPresent(cacheKey);
+            if (res == null) {
+                res = resolveEvaluationGoFeatureFlagProxy(key, defaultValue, user, expectedType);
+                cache.put(cacheKey, res);
```

Comment:
Not all the flags are cacheable.
We added a new field in the GO Feature Flag API response to know if we can or not cache the evaluation result.

What we should do here is add a new property in the file `providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/bean/GoFeatureFlagResponse.java` called `cacheable` that will tell us if we can cache the flag or not.

And here we need to check the field before being able to add it to the cache.
[You can see similar behavior in the GO provider](https://github.com/open-feature/go-sdk-contrib/blob/18d01eff5038c086ca6f2b54956513882e9e189d/providers/go-feature-flag/pkg/provider.go#L479).